javalib/src/main/scala/java/nio/channels/SelectorImpl.scala

```diff
@@ -0,0 +1,158 @@
+package java.nio.channels
+
+import java.io.IOException
+import java.net.Net
+import java.nio.channels.spi.{
+  AbstractSelectableChannel,
+  AbstractSelectionKey,
+  AbstractSelector,
+  SelectorProvider
+}
+import java.util
+import scala.collection.mutable
+import scala.scalanative.libc.stdlib.{malloc, realloc}
+import scala.scalanative.meta.LinktimeInfo.isWindows
+import scala.scalanative.unsafe.{CSize, Ptr, sizeof}
+import scala.scalanative.unsigned._
+import scalanative.posix.poll._
+import scalanative.posix.pollOps._
+
+private[channels] final class SelectorImpl(provider: SelectorProvider)
+    extends AbstractSelector(provider) {
+
+  private val keyMap = mutable.Map.empty[SelectionKey, Int]
+  private val selectedKeySet = mutable.Set.empty[SelectionKey]
+
+  private[channels] val keysLock: Object = new Object
+
+  private var pollSize: CSize = 8.toUInt
+  private var pollArr: Ptr[struct_pollfd] = malloc(
+    pollSize * sizeof[struct_pollfd]
+  ).asInstanceOf[Ptr[struct_pollfd]]
+  private var pollCount = 0
+
+  @inline private def throwIfClosed(): Unit =
+    if (!isOpen) throw new ClosedSelectorException
+
+  override protected def implCloseSelector(): Unit = ???
```

Comment:
The six "???" immediately raise a flag.  Some reviewers 
do not like to see them in mainline code.  

What to do about them depends upon planned/sketched/dreamed
next steps.  If the hope is to implement them in a next Evolution in the next few months, then mentioning that here gives an audit trail.
No sense putting a lot of work that may be soon reversed.

If they are likely to be there for an effective eternity or if you are
planning on stepping in front of a bus, the public methods should probably throw an `UnsupportedOperationException` or something
else reasonable documented for that method.

The methods returning `Unit`/void could be stubbed as either
returning `UnsupportedOperationException` or as "{}". Different
developers do either. In my private work, I prefer the Exception
so that I do not forget or lose unimplemented methods. Oops!

Just last week I ended up using the "???" in my private sandbox
work for the methods in an SPI interface class which is defined 
in Java as concrete but which is always overridden.  I had to follow the
Java specification where, if I had written the class, I would have
made it abstract from the beginning. Joys of development in
many different environments & software cultures.